### PR TITLE
Add Parquet Time32/64 conversion to CH DateTime32/64

### PR DESCRIPTION
**Pull request by arthurpassos**
### Changelog category (leave one):
- New Feature

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Add support to Parquet Time32/64 by converting it into DateTime64. Parquet time32/64 represents time elapsed since midnight, while DateTime32/64 represents an actual unix timestamp. Conversion simply offsets from `0`.

DateTime64 is used even for Time32 to leverage the `scale` attribute.

Related to #40580

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

### CODE REVIEW

**Comment by zvonand** [comment]
Tests?

File: src/Processors/Formats/Impl/ArrowColumnToCHColumn.cpp [hunk 1]
```diff
@@ -230,6 +230,28 @@ static ColumnWithTypeAndName readColumnWithTimestampData(std::shared_ptr<arrow::
     return {std::move(internal_column), std::move(internal_type), column_name};
 }
 
+static ColumnWithTypeAndName readColumnWithTimeData(std::shared_ptr<arrow::ChunkedArray> & arrow_column, const String & column_name)
+{
+    const auto & arrow_type = static_cast<const arrow::Time32Type &>(*(arrow_column->type()));
```
Comment (vdimir) [inline src/Processors/Formats/Impl/ArrowColumnToCHColumn.cpp:235]: Isn't it possible to corrupt data to have a wrong cast here? Is it validated by arrow library?

File: src/Processors/Formats/Impl/ArrowColumnToCHColumn.cpp [hunk 2]
```diff
@@ -230,6 +230,28 @@ static ColumnWithTypeAndName readColumnWithTimestampData(std::shared_ptr<arrow::
...
+    const UInt8 scale = arrow_type.unit() * 3;
+    auto internal_type = std::make_shared<DataTypeDateTime64>(scale);
+    auto internal_column = internal_type->createColumn();
+    internal_column->reserve(arrow_column->length());
+
+    for (size_t chunk_i = 0, num_chunks = static_cast<size_t>(arrow_column->num_chunks()); chunk_i < num_chunks; ++chunk_i) {
```
Comment (vdimir) [inline src/Processors/Formats/Impl/ArrowColumnToCHColumn.cpp:241]:
```suggestion
    for (size_t chunk_i = 0, num_chunks = static_cast<size_t>(arrow_column->num_chunks()); chunk_i < num_chunks; ++chunk_i)
    {
```

File: src/Processors/Formats/Impl/ArrowColumnToCHColumn.cpp (hunk 1 above)
Comment (arthurpassos) [reply src/Processors/Formats/Impl/ArrowColumnToCHColumn.cpp]: Nice catch, I have separated it into two functions that use appropriate types. Can you enable CI?

Comment (vdimir) [reply src/Processors/Formats/Impl/ArrowColumnToCHColumn.cpp]: Oh, I meant a bit different case, but it's good that we found it :) When type tag is arrow::Type::TIME32, but actual type would be different from `arrow::Time32Type` is it possible to corrupt data in such way?

Also it's good to add tests-case to cover new code. I've added can-be-tested label

Comment (arthurpassos) [reply src/Processors/Formats/Impl/ArrowColumnToCHColumn.cpp]: Hm.. I am not exactly sure, tbh. What I can say is that there is already code that does exactly the same, see parquet timestamp conversion [example](https://github.com/ClickHouse/ClickHouse/blob/b26cc5a1ba791df03e56117c6207aec97b25b58b/src/Processors/Formats/Impl/ArrowColumnToCHColumn.cpp#L215).

I'll be adding tests in a few.

Comment (arthurpassos) [reply src/Processors/Formats/Impl/ArrowColumnToCHColumn.cpp]: I'll actually set this as draft while I don't have the tests

Comment (arthurpassos) [reply src/Processors/Formats/Impl/ArrowColumnToCHColumn.cpp]: Open for review again :)

File: tests/queries/0_stateless/00900_parquet_time_to_ch_date_time.sh [hunk 3]
```diff
@@ -0,0 +1,25 @@
+#!/usr/bin/env bash
```
Comment (vdimir) [inline tests/queries/0_stateless/00900_parquet_time_to_ch_date_time.sh:1]:
```suggestion
#!/usr/bin/env bash
# Tags: no-fasttest
```

**Comment by vdimir** [comment]
> Stateless tests (release)

 https://github.com/ClickHouse/ClickHouse/pull/41501

**Review by vdimir — COMMENTED** [review]
Merged, thank you for your contribution!

**Pull request merged**